crates/sim2h/src/cache.rs

```diff
@@ -58,6 +74,41 @@ impl Space {
         self.missing_aspects.keys().cloned().collect()
     }
 
+    /// Returns true if the given agent is missing all of the given aspects for the given entry.
+    /// That is: if all of the aspects are recorded as missing for that agent.
+    /// If one of the given aspects is not in that vector of missing entries, the agent is supposed
+    /// to have it and this function returns false.
+    pub fn agent_is_missing_all_aspects(
+        &self,
+        agent_id: &AgentId,
+        entry_hash: &EntryHash,
+        aspects: &Vec<AspectHash>,
+    ) -> bool {
+        let maybe_agent_map = self.missing_aspects.get(agent_id);
+        if maybe_agent_map.is_none() {
+            return false;
+        }
+        let map_for_agent = maybe_agent_map.unwrap();
+
+        let maybe_vec_of_missing_aspects_for_entry = map_for_agent.get(entry_hash);
+        if maybe_vec_of_missing_aspects_for_entry.is_none() {
+            return false;
+        }
+
+        let missing_aspects_for_entry = maybe_vec_of_missing_aspects_for_entry.unwrap();
+
+        // We check that every of the given aspects is the missing list.
+        // If one is missing from the missing list this block returns some
+        // and the whole function returns false.
+        for aspect in aspects {
+            if !missing_aspects_for_entry.contains(aspect) {
+                return false;
+            }
+        }
```

Comment:
FYI we could use [`HashSet::difference()`](https://doc.rust-lang.org/std/collections/struct.HashSet.html#method.difference) for this, but I'm not sure if the speed gain is worth it